src/tools/kdb/CMakeLists.txt

```diff
@@ -7,27 +7,33 @@ add_toolheaders (HDR_FILES)
 include_directories (${CMAKE_CURRENT_SOURCE_DIR})
 
 file (GLOB SRC_FILES *.cpp)
-
 set (SOURCES ${SRC_FILES} ${HDR_FILES})
 
+if (BUILD_SHARED OR BUILD_FULL)
+    add_library (kdb-objects OBJECT ${SRC_FILES})
+endif (BUILD_SHARED OR BUILD_FULL)
+
 if (BUILD_SHARED)
-	add_executable (kdb ${SOURCES})
+    add_executable (kdb $<TARGET_OBJECTS:kdb-objects>)
+    add_dependencies (kdb kdberrors_generated)
 
-	target_link_libraries (kdb elektra-core elektra-kdb elektratools)
+    target_link_libraries (kdb elektra-core elektra-kdb elektratools)
 
-	install (TARGETS kdb DESTINATION bin)
+    install (TARGETS kdb DESTINATION bin)
 endif (BUILD_SHARED)
 
 if (BUILD_FULL)
-	add_executable (kdb-full ${SOURCES})
+    add_executable (kdb-full $<TARGET_OBJECTS:kdb-objects>)
+    add_dependencies (kdb-full kdberrors_generated)
 
-	target_link_libraries (kdb-full elektra-full elektratools-full)
+    target_link_libraries (kdb-full elektra-full elektratools-full)
 
-	install (TARGETS kdb-full DESTINATION bin)
+    install (TARGETS kdb-full DESTINATION bin)
 endif (BUILD_FULL)
 
 if (BUILD_STATIC)
 	add_executable (kdb-static ${SOURCES})
+	add_dependencies (kdb-static kdberrors_generated)
```

Comment:
Ok, maybe the branch you used for this PR is too old. It might be the easiest option to branch off from master again and overwrite the CMakeLists.txt with the correct content.